# Activity 3.1 Introduction to Data Analysis and Tools
#### by: John Edward Binay
#### Submitted on: 11-27-2021

Intended Learning Outcome

1. Use pandas and numpy data analysis tools.

2. Demonstrate how to analyze data using numpy and pandas

Resources:

* Personal Computer

* Jupyter Notebook

* Internet Connection
    

Task:

Load the diabetes.csv file. Convert the diabetes.csv into dataframe



Perform the following tasks in the diabetes dataframe:

1. Identify the column names 
2. Identify the data types of the data 
3. Display the total number of records
4. Display the first 20 records
5. Display the last 20 records
6. Change the Outcome column to Diagnosis
7. Create a new column Classification that display "Diabetes" if the value of outcome is 1 , otherwise "No Diabetes"
8. Create a new dataframe "withDiabetes" that gathers data with diabetes 
9. Create a new dataframe "noDiabetes" thats gathers data with no diabetes
10. Create a new dataframe "Pedia" that gathers data with age 0 to 19
11. Create a new dataframe "Adult" that gathers data with age greater than 19
12. Use numpy to get the average age and glucose value.
13. Use numpy to get the median age and glucose value.
14. Use numpy to get the middle values of glucose and age.
15. Use numpy to get the standard deviation of the skinthickness.

In [ ]:
# load the diabetes.csv file
# import numpy and pandas

import pandas as pd
import numpy as np

from google.colab import drive
drive.mount('/content/drive')
filepath = '/content/drive/MyDrive/Colab Notebooks/diabetes.csv'
data = pd.read_csv(filepath)

In [16]:
# Identify the column names
data



Pregnancies  Glucose  ...  Age  Outcome
0              6      148  ...   50        1
1              1       85  ...   31        0
2              8      183  ...   32        1
3              1       89  ...   21        0
4              0      137  ...   33        1
..           ...      ...  ...  ...      ...
763           10      101  ...   63        0
764            2      122  ...   27        0
765            5      121  ...   30        0
766            1      126  ...   47        1
767            1       93  ...   23        0

[768 rows x 9 columns]

In [17]:
# Identify the cilumn names pt. 2
print(data.columns.values)

['Pregnancies' 'Glucose' 'BloodPressure' 'SkinThickness' 'Insulin' 'BMI'
 'DiabetesPedigreeFunction' 'Age' 'Outcome']


In [23]:
# Identify the data types of the data

dataType = data.dtypes
print('Data type of each column of Dataframe :\n')
print(dataType)

Data type of each column of Dataframe :

Pregnancies                   int64
Glucose                       int64
BloodPressure                 int64
SkinThickness                 int64
Insulin                       int64
BMI                         float64
DiabetesPedigreeFunction    float64
Age                           int64
Outcome                       int64
dtype: object


In [24]:
# Display the total number of records

data.shape[0]

768

In [28]:
# Display the first 20 records
data[:20] # or: data.head(20)

Pregnancies  Glucose  BloodPressure  ...  DiabetesPedigreeFunction  Age  Outcome
0             6      148             72  ...                     0.627   50        1
1             1       85             66  ...                     0.351   31        0
2             8      183             64  ...                     0.672   32        1
3             1       89             66  ...                     0.167   21        0
4             0      137             40  ...                     2.288   33        1
5             5      116             74  ...                     0.201   30        0
6             3       78             50  ...                     0.248   26        1
7            10      115              0  ...                     0.134   29        0
8             2      197             70  ...                     0.158   53        1
9             8      125             96  ...                     0.232   54        1
10            4      110             92  ...                     0.191   30        0
11           10      168             74  ...                     0.537   34        1
12           10      139             80  ...                     1.441   57        0
13            1      189             60  ...                     0.398   59        1
14            5      166             72  ...                     0.587   51        1
15            7      100              0  ...                     0.484   32        1
16            0      118             84  ...                     0.551   31        1
17            7      107             74  ...                     0.254   31        1
18            1      103             30  ...                     0.183   33        0
19            1      115             70  ...                     0.529   32        1

[20 rows x 9 columns]

In [33]:
# Display the last 20 records
data.tail(20)

Pregnancies  Glucose  ...  Age  Outcome
748            3      187  ...   36        1
749            6      162  ...   50        1
750            4      136  ...   22        1
751            1      121  ...   28        0
752            3      108  ...   25        0
753            0      181  ...   26        1
754            8      154  ...   45        1
755            1      128  ...   37        1
756            7      137  ...   39        0
757            0      123  ...   52        1
758            1      106  ...   26        0
759            6      190  ...   66        1
760            2       88  ...   22        0
761            9      170  ...   43        1
762            9       89  ...   33        0
763           10      101  ...   63        0
764            2      122  ...   27        0
765            5      121  ...   30        0
766            1      126  ...   47        1
767            1       93  ...   23        0

[20 rows x 9 columns]

In [35]:
# Change the Outcome column to Diagnosis
data.rename(columns ={'Outcome' : 'Diagnosis'}, inplace = True)
data

Pregnancies  Glucose  ...  Age  Diagnosis
0              6      148  ...   50          1
1              1       85  ...   31          0
2              8      183  ...   32          1
3              1       89  ...   21          0
4              0      137  ...   33          1
..           ...      ...  ...  ...        ...
763           10      101  ...   63          0
764            2      122  ...   27          0
765            5      121  ...   30          0
766            1      126  ...   47          1
767            1       93  ...   23          0

[768 rows x 9 columns]

In [39]:
# Create a new column Classification that display "Diabetes" if the value of Diagnosis is 1 , 
# otherwise "No Diabetes"

data['Classification'] = np.where(data['Diagnosis'] == 1, 'Diabetes', 'No Diabetes')
data

Pregnancies  Glucose  BloodPressure  ...  Age  Diagnosis  Classification
0              6      148             72  ...   50          1        Diabetes
1              1       85             66  ...   31          0     No Diabetes
2              8      183             64  ...   32          1        Diabetes
3              1       89             66  ...   21          0     No Diabetes
4              0      137             40  ...   33          1        Diabetes
..           ...      ...            ...  ...  ...        ...             ...
763           10      101             76  ...   63          0     No Diabetes
764            2      122             70  ...   27          0     No Diabetes
765            5      121             72  ...   30          0     No Diabetes
766            1      126             60  ...   47          1        Diabetes
767            1       93             70  ...   23          0     No Diabetes

[768 rows x 10 columns]

In [43]:
# Create a new dataframe "withDiabetes" that gathers data with diabetes

withDiabetes = data[(data['Diagnosis'] ==  1)]
withDiabetes

Pregnancies  Glucose  BloodPressure  ...  Age  Diagnosis  Classification
0              6      148             72  ...   50          1        Diabetes
2              8      183             64  ...   32          1        Diabetes
4              0      137             40  ...   33          1        Diabetes
6              3       78             50  ...   26          1        Diabetes
8              2      197             70  ...   53          1        Diabetes
..           ...      ...            ...  ...  ...        ...             ...
755            1      128             88  ...   37          1        Diabetes
757            0      123             72  ...   52          1        Diabetes
759            6      190             92  ...   66          1        Diabetes
761            9      170             74  ...   43          1        Diabetes
766            1      126             60  ...   47          1        Diabetes

[268 rows x 10 columns]

In [54]:
# Create a new dataframe "Pedia" that gathers data with age 0 to 19

Pedia = data[(data['Age'] >=  0) & (data['Age'] <=  19)]
Pedia

#! a better way of doing this is by using the code:
#? data[np.logical_and(data['Age'] >= 0, data["Age"] <= 19)]

Empty DataFrame
Columns: [Pregnancies, Glucose, BloodPressure, SkinThickness, Insulin, BMI, DiabetesPedigreeFunction, Age, Diagnosis, Classification]
Index: []

In [55]:
# Create a new dataframe "Adult" that gathers data with age greater than 19

Adult = data[(data['Age'] >  19)]
Adult


Pregnancies  Glucose  BloodPressure  ...  Age  Diagnosis  Classification
0              6      148             72  ...   50          1        Diabetes
1              1       85             66  ...   31          0     No Diabetes
2              8      183             64  ...   32          1        Diabetes
3              1       89             66  ...   21          0     No Diabetes
4              0      137             40  ...   33          1        Diabetes
..           ...      ...            ...  ...  ...        ...             ...
763           10      101             76  ...   63          0     No Diabetes
764            2      122             70  ...   27          0     No Diabetes
765            5      121             72  ...   30          0     No Diabetes
766            1      126             60  ...   47          1        Diabetes
767            1       93             70  ...   23          0     No Diabetes

[768 rows x 10 columns]

In [58]:
# Use numpy to get the average age and glucose value.
print('the average age in the data is', data.Age.mean())
print('the average glucose in the data is', data.Glucose.mean())

the average age in the data is 33.240885416666664
the average glucose in the data is 120.89453125


In [59]:
# Use numpy to get the median age and glucose value.

print('the median age in the data is', data.Age.median())
print('the median glucose in the data is', data.Glucose.median())

the median age in the data is 29.0
the median glucose in the data is 117.0


In [60]:
# Use numpy to get the middle values of glucose and age.
# Since the median value of a set is the same as the middel value of that set, then 
# findin the median (as what we did in the previous exercise) is the correct solution

print('the middle value in column Age in the data is', data.Age.median())
print('the middle value in column glucose in the data is', data.Glucose.median())

the middle value in column Age in the data is 29.0
the middle value in column glucose in the data is 117.0


In [61]:
# Use numpy to get the standard deviation of the skinthickness.

print("Standard Deviation of skinthickness:", data.SkinThickness.std())

Standard Deviation of skinthickness: 15.952217567727677


Conclusion

The libraries, Pandas and Numpy, are super intersting and powerful tools to use in manipulating, displaying, and finding out statistical importance of any given data. They handle .csv files very well. They also allow the user to create and manipulate matrices very easily. The heart of data science computations and data structures rely on matrices and matrice operations. These two libraries help acheive that. I can say that I perosnally enjoyed working with data for the first time. It also opened my eyes on the ways that I can maniupulte large data sets. The file, diabetes.csv, has about 768 rows/entries. It would be difficult to filter, manipualte, and understand these data without using python and the libraries of pandas and numpy.